Sma 48 main - linting for test cases and scripts

livingrockrises opened the pull request.

Aboudjem commented on .eslintrc.js:
I recommend adding these rules:

```
  "@typescript-eslint/no-explicit-any": "error",
  "quotes": ["error", "double"],
  "no-unused-vars": "error",
  "no-console": "error",
  "@typescript-eslint/no-unused-vars": "error",
  "@typescript-eslint/explicit-function-return-type": "error"

```

Aboudjem replied:
I suggest we use this security tool for ESLint: [eslint-plugin-security](https://github.com/eslint-community/eslint-plugin-security). It's good for keeping our code safe probably not useful here but it can be important for all our projects, not just this one.

Aboudjem replied:
We can fix in a future PR since it requires a lot of time

Aboudjem commented on package.json:
I'd like to propose some changes to our linting scripts in package.json:

```
{
  "lint:ts": "npx eslint '**/*.ts'",
  "lint:ts-fix": "npx eslint '**/*.ts' --fix",
  "lint:sol": "npx prettier '**/*.{json,sol,md}' --check && npx solhint 'contracts/**/*.sol'",
  "lint:sol-fix": "npx prettier '**/*.{json,sol,md}' --write && npx solhint 'contracts/**/*.sol' --fix",
  "lint": "npm run lint:ts && npm run lint:sol",
  "lint:fix": "npm run lint:ts-fix && npm run lint:sol-fix"
}
```

- Using `:` in the script names like lint:sol helps us quickly understand that it's a linting script for Solidity. It follows the pattern `lint:language-action`, making knowing what each script does easily

- Added scripts for Checks only, it only checks without fixing (lint, lint:sol and lint:ts) it will be useful for our CI/CD pipeline

Aboudjem reviewed (changes requested):
minor changes

livingrockrises commented:
1. Added below rules, but at the moment I have kept all of them commented out

    ```
     // "@typescript-eslint/no-explicit-any": "error",
    // quotes: ["error", "double"],
    // "no-unused-vars": "error",
    // "no-console": "error",
    // "@typescript-eslint/no-unused-vars": "error",
    // "@typescript-eslint/explicit-function-return-type": "error",
```

2. package.json scripts look like this now.
```

    "lint:ts": "npx eslint 'test/**/*.ts' 'scripts/**/*.ts'",
    "lint:ts-fix": "npx eslint 'test/**/*.ts' 'scripts/**/*.ts' --fix",
    "lint:sol": "npx solhint 'contracts/**/*.sol'",
    "lint:sol-fix": "npx prettier 'contracts/**/*.sol' --write && npx solhint 'contracts/**/*.sol' --fix",
    "lint": "npm run lint:ts && npm run lint:sol",
    "lint:fix": "npm run lint:ts-fix && npm run lint:sol-fix",
```

 2a. I have limited linting to scripts and test folders for now, as other folders under src/ don't pass for existing eslint rules (will need to be reviewed and updated all together - this PR being lint fixes for test cases mainly)
 
 2b. lint:sol doesnt need to check json, md files etc so it's rn limited to solidity files.
